nodes/config/ui_base.html

```diff
@@ -212,10 +237,71 @@
         },
         onpaletteadd: function () {
             // add the Dashboard 2.0 sidebar
+            if (RED._db2debug) { console.log('dashboard 2: ui_base.html: onpaletteadd ()') }
             addSidebar()
         }
     })
 
+    /**
+     * Utility function to convert a dashboard node to a DashboardItem
+     * Provides a better DX with type checking and intellisense
+     * @param {Object} node - The node to convert
+     * @returns {DashboardItem}
+     */
+    function toDashboardItem (node) {
+        if (RED._db2debug) { console.log('dashboard 2: ui_base.html: toDashboardItem (node)', node) }
+        /** @type {DashboardItem} */
+        const item = {
+            itemType: 'widget',
+            id: node.id,
+            name: node.name,
+            type: node.type,
+            order: node.order,
+            label: null,
+            icon: null,
+            color: null,
+            isSubflowInstance: false,
+            node
+        }
+        if (hasProperty(node, 'group')) { item.group = node.group }
+        if (hasProperty(node, 'page')) { item.page = node.page }
+        if (hasProperty(node, 'theme')) { item.theme = node.theme }
+        if (hasProperty(node, 'env') && Array.isArray(node.env) && /subflow:.+/.test(node.type)) {
+            const envOrder = node.env.find(e => e.key === 'DB2_SF_ORDER')
+            if (envOrder) {
+                item.order = envOrder.value
+            }
+        }
+        switch (node.type) {
+        case 'ui-page':
+        case 'ui-group':
+        case 'ui-theme':
+        case 'ui-base':
+            item.itemType = node.type.replace('ui-', '')
+            break
+        default:
+            item.itemType = 'widget'
+            break
+        }
+        try {
+            item.order = parseInt(item.order)
+        } finally {
+            item.order = isNaN(item.order) ? 0 : item.order
+        }
+        try {
+            item.label = RED.utils.getNodeLabel(node)
+        } finally {
+            item.label = node.label || node.name || node.id
+        }
+        try {
+            item.icon = RED.utils.getNodeIcon(node)
```

Comment:
why is `getNodeIcon`? Why is this even required in the `widget` object?

Reply:
This might be redundant now. I'll double check. The intention was to show the respective icon in the editableList for clearer representation. 

And yes, subflows can have an icon chosen by the user.